crates/telio-wg/src/link_detection.rs

```diff
@@ -240,75 +216,170 @@ pub struct LinkDetectionUpdateResult {
     pub link_state: Option<LinkState>,
 }
 
-#[derive(Clone, Copy, Debug)]
-struct NodeInfo {
-    last_tx_timestamp: Instant,
-    last_rx_timestamp: Instant,
-    last_tx_bytes: u64,
-    last_rx_bytes: u64,
-    last_link_state: LinkState,
+enum State {
+    Down {
+        stats: BytesAndTimestamps,
+    },
+    PossibleDown {
+        stats: BytesAndTimestamps,
+        count: usize,
+    },
+    Up {
+        stats: BytesAndTimestamps,
+    },
 }
```

Comment:
IMHO would be nicer to do something like this:
```suggestion
struct State {
    stats: BytesAndTimestamps,
    variant: StateVariant,
}
enum StateVariant {
    Down,
    PossibleDown {
        count: usize,
    },
    Up,
}
```
It may reduce noise in the `update()` function

Reply:
Good idea <3 I did it